subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -671,3 +740,27 @@ void main(void)
 	}
 }
 #endif
+
+#ifdef CONFIG_ARCH_POSIX
+static void add_test_filter_option(void)
+{
+	static struct args_struct_t test_filter_s[] = {
+		/*
+		 * Fields:
+		 * manual, mandatory, switch,
+		 * option_name, var_name ,type,
+		 * destination, callback,
+		 * description
+		 */
+		{ false, false, false, "test", "test name", 's', (void *)&test_filter, NULL,
+		  "Name of test to run" },
+		{ false, false, false, "suite", "suite name", 's', (void *)&suite_filter, NULL,
+		  "Name of suite to run" },
+		ARG_TABLE_ENDMARKER
+	};
+
+	native_add_command_line_opts(test_filter_s);
+}
+
+NATIVE_TASK(add_test_filter_option, PRE_BOOT_1, 10);
+#endif
```

Comment:
Can you move this to another PR. I think we need to be able to re-run multiple tests/suites. Also, keep in mind that the test name doesn't need to be unique, instead it is the combination of suite/test that needs to be unique. I think this extra feature will slow down merging this PR.

Reply:
Moved to https://github.com/zephyrproject-rtos/zephyr/pull/44382